tests/SeekTest.php

```diff
@@ -99,35 +100,53 @@ public function testMoveLast()
         $actual = $this->db->SeekPosition();
 
         $this->assertSame($expected, $actual);
+        $this->assertFalse($this->db->EndOfSeek());
     }
-    
+
     public function testBeginningOfSeekNoConnection()
     {
         $this->db->Close();
         $this->db->Seek(0);
-        $this->assertFalse($this->db->BeginningOfSeek());        
+        $this->assertFalse($this->db->BeginningOfSeek());
     }
-    
+
     public function testEndOfSeekNoConnection()
     {
         $this->db->Close();
         $this->db->Seek(0);
-        $this->assertFalse($this->db->EndOfSeek());        
-    }  
-    
+        $this->assertFalse($this->db->EndOfSeek());
+    }
+
     public function testMoveFirstNoConnection()
     {
         $this->db->Close();
         $this->db->Query("SELECT * FROM `test_table`");
         $this->db->Seek(0);
-        $this->assertFalse($this->db->MoveFirst());        
+        $this->assertFalse($this->db->MoveFirst());
     }
-    
+
     public function testMoveLastNoConnection()
     {
         $this->db->Close();
         $this->db->Query("SELECT * FROM `test_table`");
         $this->db->Seek(0);
         $this->assertFalse($this->db->MoveLast());
     }
+
+    /**
+     * Test based on some real world code which worked in version 3.0
+     * and was broken with release 4.0.
+     */
+    public function testSeekLoop()
+    {
+        $this->db->Query("SELECT * FROM `test_table`");
+        $i = 0;
+        $this->db->MoveFirst();
+        while (!$this->db->EndOfSeek()) {
+            $this->db->Row();
+            $i++;
+        }
+        $this->assertSame(2, $i);
```

Comment:
This must be 3, because you have added 1 row

Reply:
Yep. Adapted to the other test data change.